Formula/tbb.rb

```diff
@@ -81,6 +84,29 @@ def install
     assert_path_exists lib/"libtbb.a"
     assert_path_exists lib/"libtbbmalloc.a"
 
+    # on macOS core types are not distinguished, because libhwloc does not support it for now
+    # see https://github.com/oneapi-src/oneTBB/blob/690aaf497a78a75ff72cddb084579427ab0a8ffc/CMakeLists.txt#L226-L228
+    (testpath/"cores-types.cpp").write <<~EOS
+      #include <iostream>
+      #include <tbb/task_arena.h>
+
+      int main() {
+          const auto numa_nodes = tbb::info::numa_nodes();
+          const auto size = numa_nodes.size();
+          const auto type = numa_nodes.front();
+      #ifdef __APPLE__
+          std::cout << (size == 1 && type == tbb::task_arena::automatic ? EXIT_SUCCESS : EXIT_FAILURE);
+      #else
+          std::cout << (size != 1 || type != tbb::task_arena::automatic ? EXIT_SUCCESS : EXIT_FAILURE);
```

Comment:
Perhaps we can `return` these values instead of printing them to stdout? That way we can just do
```ruby
system "./core-types"
```
instead of reading the output with `assert_equal`.

Reply:
great suggestion, updated.